internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -269,539 +226,312 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
-		openAPISchema,
-		opts...,
-	)
-	defer cleanup()
-	if err != nil {
-		return gitopsSyncCommon.OperationError, err
+		return false, false, err
 	}
 
-	syncCtx.Sync()
-
-	rollout.Status.MarkDeployed(rollout.Generation)
+	NumaflowControllerNeedsToUpdate := !reflect.DeepEqual(existingSpecAsMap, newSpecAsMap)
 
-	phase, _, _ := syncCtx.GetState()
-	return phase, nil
+	return NumaflowControllerNeedsToUpdate, !numaflowControllerReconciled, nil
 }
 
-// compareState compares with desired state of the objects with the live state in the cluster
-// for the target objects.
-func (r *NumaflowControllerRolloutReconciler) compareState(
-	rollout *apiv1.NumaflowControllerRollout,
-	namespace string,
-	targetObjs []*unstructured.Unstructured,
-	numaLogger *logger.NumaLogger,
-) (gitopsSync.ReconciliationResult, *diff.DiffResultList, error) {
-	var infoProvider kubeUtil.ResourceInfoProvider
-	clusterCache, err := r.stateCache.GetClusterCache()
-	if err != nil {
-		return gitopsSync.ReconciliationResult{}, nil, err
-	}
-	infoProvider = clusterCache
-	liveObjByKey, err := r.stateCache.GetManagedLiveObjs(rollout.Name, namespace, targetObjs)
-	if err != nil {
-		return gitopsSync.ReconciliationResult{}, nil, err
-	}
-	reconciliationResult := gitopsSync.Reconcile(targetObjs, liveObjByKey, namespace, infoProvider)
-
-	// Ignore `status` field for all comparison.
-	// TODO: make it configurable
-	overrides := map[string]sync.ResourceOverride{
-		"*/*": {
-			IgnoreDifferences: sync.OverrideIgnoreDiff{JSONPointers: []string{"/status"}}},
-	}
+// TODO: do we need to also check the underlying Deployment?
+// determine if the NumaflowController, including its underlying Deployment, has been reconciled
+// so, this requires:
+// 1. NumaflowController.Status.ObservedGeneration == NumaflowController.Generation
+// 2. Deployment.Status.ObservedGeneration == Deployment.Generation
+// 3. Deployment.Status.UpdatedReplicas == Deployment.Spec.Replicas
+func (r *NumaflowControllerRolloutReconciler) isNumaflowControllerReconciled(ctx context.Context, numaflowController *unstructured.Unstructured) (bool, string, error) {
+	numaLogger := logger.FromContext(ctx)
 
-	resourceOps, cleanup, err := r.getResourceOperations()
+	numaflowControllerStatus, err := kubernetes.ParseStatus(numaflowController)
 	if err != nil {
-		return gitopsSync.ReconciliationResult{}, nil, err
+		return false, "", fmt.Errorf("failed to parse Status from NumaflowController CR: %+v, %v", numaflowController, err)
 	}
-	defer cleanup()
+	numaLogger.Debugf("numaflowController status: %+v", numaflowControllerStatus)
 
-	diffOpts := []diff.Option{
-		diff.WithLogr(*numaLogger.LogrLogger),
-		diff.WithServerSideDiff(true),
-		diff.WithServerSideDryRunner(diff.NewK8sServerSideDryRunner(resourceOps)),
-		diff.WithManager(common.SSAManager),
-		diff.WithGVKParser(clusterCache.GetGVKParser()),
-	}
+	numaflowControllerReconciled := numaflowController.GetGeneration() <= numaflowControllerStatus.ObservedGeneration
```

Comment:
so, I'm guessing this one isn't working either right now...but that can be handled separately I guess

Reply:
Good catch. Once we get in your temporary fix on the unstructured util this should resolve as well.